synapse/util/caches/lrucache.py

```diff
@@ -76,14 +78,19 @@ def __init__(
         """
         cache = cache_type()
         self.cache = cache  # Used for introspection.
+
+        # Save the original max size, and apply the default size factor.
+        self._original_max_size = max_size
+        self.max_size = math.floor(max_size * cache_config.DEFAULT_CACHE_SIZE_FACTOR)
```

Comment:
Any particular reason to use `math.floor` instead of truncating with `int`?

Reply:
ehhh `int()` coersion feels icky :)